importers/algod/algod_importer.go

```diff
@@ -0,0 +1,88 @@
+package algodimporter
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/plugins"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+const importerName = "algod"
+
+type algodImporter struct {
+	aclient *algod.Client
+	logger  *logrus.Logger
+	cfg     ImporterConfig
+	ctx     context.Context
+	cancel  context.CancelFunc
+}
+
+var algodImporterMetadata = importers.ImporterMetadata{
+	ImpName:        importerName,
+	ImpDescription: "Importer for fetching block from algod rest endpoint.",
+	ImpDeprecated:  false,
+}
+
+// New initializes an algod importer
+func New() importers.Importer {
+	return &algodImporter{}
+}
+
+func (bot *algodImporter) Metadata() importers.ImporterMetadata {
+	return algodImporterMetadata
+}
+
+func (bot *algodImporter) Init(ctx context.Context, cfg plugins.PluginConfig, logger *logrus.Logger) (err error) {
+	bot.ctx, bot.cancel = context.WithCancel(ctx)
+	bot.logger = logger
+	if err := bot.unmarhshalConfig(string(cfg)); err != nil {
+		return fmt.Errorf("connect failure in unmarshalConfig: %v", err)
+	}
+	var client *algod.Client
+	u, err := url.Parse(bot.cfg.NetAddr)
+	if err != nil {
+		return
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		bot.cfg.NetAddr = "http://" + bot.cfg.NetAddr
+	}
+	client, err = algod.MakeClient(bot.cfg.NetAddr, bot.cfg.Token)
+	if err != nil {
+		return
+	}
+	bot.aclient = client
+	return
+}
+
+func (bot *algodImporter) Close() (err error) {
+	bot.cancel()
+	bot = nil
```

Comment:
I've never seen this before. What is the purpose of doing this?

Reply:
`bot = nil` is unnecessary, will remove it